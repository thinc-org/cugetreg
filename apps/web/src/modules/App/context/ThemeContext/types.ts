export type ThemeProviderProps = React.PropsWithChildren<{ forceDark?: boolean }>

export type ThemeType = 'Light' | 'Dark'

export type IThemeContext = {
  type: ThemeType
  toggleType: () => void
}
