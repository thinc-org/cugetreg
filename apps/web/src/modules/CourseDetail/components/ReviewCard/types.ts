import { Review } from '@web/common/types/reviews'

export type ReviewCardProps = Review
