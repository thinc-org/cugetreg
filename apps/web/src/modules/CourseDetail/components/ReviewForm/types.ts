import { Review } from '@web/common/types/reviews'
import { ReviewState } from '@web/modules/CourseDetail/context/Review/types'

export type ReviewEditables = Pick<Review, 'rating' | 'academicYear' | 'semester' | 'content'>

export interface ReviewFormRef {
  clearForm: () => void
  storeLocalReviewForm: () => void
  restoreFormState: (form: Partial<ReviewState>) => void
  applyFromReview: (review: Review) => void
  toReview: () => ReviewEditables
}
