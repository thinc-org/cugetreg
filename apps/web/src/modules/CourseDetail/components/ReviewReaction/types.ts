import { ReviewInteractionType } from '@libs/codegen'

export interface ReviewReactionProps {
  type: ReviewInteractionType
  pressed: boolean
  reactionCount: number
  onClick?: () => void
}
