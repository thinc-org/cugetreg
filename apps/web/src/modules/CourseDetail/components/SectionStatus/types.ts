import { TypographyProps } from '@mui/material'

import { Capacity } from '@libs/codegen'

export type AvailableStatus = 'available' | 'full' | 'closed'
export interface SectionStatusProps extends TypographyProps {
  capacity: Capacity
  closed: boolean
}
