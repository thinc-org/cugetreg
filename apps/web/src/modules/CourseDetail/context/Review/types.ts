import { RefObject } from 'react'

import { TNode } from '@udecode/plate-core'
import { ReviewFormRef } from '@web/modules/CourseDetail/components/ReviewForm/types'

import { Review } from '@cgr/codegen'
import { ReviewInteractionType } from '@cgr/codegen'

export type ReviewState = Pick<Review, 'rating' | 'academicYear' | 'semester'> & {
  content: TNode[]
}

export interface ReviewContextValues {
  courseNo: string
  reviews: Review[]
  myPendingReviews: Review[]
  setInteraction: (reviewId: string, interaction: ReviewInteractionType) => void
  reportReview: (reviewId: string) => void
  deleteMyReview: (reviewId: string) => void
  editMyReview: (reviewId: string) => void
  cancelEditReview: () => void
  submitReview: () => void
  submitEditedReview: (reviewId: string) => void
  editingReviewId?: string
  formRef: RefObject<ReviewFormRef | undefined>
  formLoaded: boolean
  onFormLoad: () => void
}

export interface ReviewProviderProps {
  courseNo: string
  initialReviews: Review[]
  children?: React.ReactNode
}
