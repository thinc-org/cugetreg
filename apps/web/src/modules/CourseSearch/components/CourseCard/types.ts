import { Course } from '@libs/codegen'

export interface CourseCardProps {
  course: Course
  rating?: number
}
