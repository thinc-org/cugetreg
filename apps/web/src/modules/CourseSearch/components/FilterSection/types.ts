export interface FilterSectionProps {
  open: boolean
  handleClose: () => void
}
