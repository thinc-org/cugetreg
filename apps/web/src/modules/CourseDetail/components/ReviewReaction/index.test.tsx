import { MdThumbDown, MdThumbDownOffAlt, MdThumbUp, MdThumbUpOffAlt } from 'react-icons/md'

import { ThemeProvider } from '@mui/material'
import { lightTheme } from '@web/configs/theme'
import { shallow } from 'enzyme'

import { ReviewInteractionType } from '@cgr/codegen'

import { ReviewReactionProps } from './types'

describe('ReviewReaction', () => {
  const mockProps: ReviewReactionProps = {
    type: ReviewInteractionType.L,
    pressed: false,
    reactionCount: 42,
  }

  const useThemeSpy = jest.fn(() => lightTheme)

  jest.doMock('@mui/material', () => ({
    ...(jest.requireActual('@mui/material') as any),
    useTheme: useThemeSpy,
  }))

  afterEach(() => {
    jest.clearAllMocks()
  })

  it.each`
    type                       | defaultPressed | expectedIcon
    ${ReviewInteractionType.L} | ${false}       | ${MdThumbUpOffAlt}
    ${ReviewInteractionType.D} | ${false}       | ${MdThumbDownOffAlt}
    ${ReviewInteractionType.L} | ${true}        | ${MdThumbUp}
    ${ReviewInteractionType.D} | ${true}        | ${MdThumbDown}
  `(
    'should render ReviewReaction type $type like correctly when defaultPressed=$defaultPressed ',
    async ({ type, expectedIcon, defaultPressed }) => {
      const { ReviewReaction } = await import('.')
      const wrapper = shallow(
        <ReviewReaction {...mockProps} type={type} pressed={defaultPressed} />,
        {
          wrappingComponent: ThemeProvider,
          wrappingComponentProps: { theme: lightTheme },
        }
      )
      expect(wrapper.find(expectedIcon)).toBeTruthy()
      expect(wrapper).toMatchSnapshot()
    }
  )

  it.each`
    type                       | expectedUnpressedIcon | expectedPressedIcon
    ${ReviewInteractionType.L} | ${MdThumbUpOffAlt}    | ${MdThumbUp}
    ${ReviewInteractionType.D} | ${MdThumbDownOffAlt}  | ${MdThumbDown}
  `(
    'should change icon when the component was clicked, type=$type',
    async ({ type, expectedUnpressedIcon, expectedPressedIcon }) => {
      const { ReviewReaction } = await import('.')
      const wrapper = shallow(<ReviewReaction {...mockProps} type={type} />, {
        wrappingComponent: ThemeProvider,
        wrappingComponentProps: { theme: lightTheme },
      })
      expect(wrapper.find(expectedUnpressedIcon)).toBeTruthy()
      wrapper.simulate('click')
      expect(wrapper.find(expectedPressedIcon)).toBeTruthy()
      wrapper.simulate('click')
      expect(wrapper.find(expectedUnpressedIcon)).toBeTruthy()
    }
  )
})
