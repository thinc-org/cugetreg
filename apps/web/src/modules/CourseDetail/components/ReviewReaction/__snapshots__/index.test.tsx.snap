// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReviewReaction > should render ReviewReaction type 'D' like correctly when defaultPressed=false  1`] = `
<ForwardRef(Grid)
  alignItems="center"
  direction="row"
  spacing={0.5}
>
  <ForwardRef(IconButton)
    size="small"
  >
    <SelectedIcon />
  </ForwardRef(IconButton)>
  <ForwardRef(Typography)
    color="#C9CCE4"
    variant="h6"
  >
    42
  </ForwardRef(Typography)>
</ForwardRef(Grid)>
`;

exports[`ReviewReaction > should render ReviewReaction type 'D' like correctly when defaultPressed=true  1`] = `
<ForwardRef(Grid)
  alignItems="center"
  direction="row"
  spacing={0.5}
>
  <ForwardRef(IconButton)
    size="small"
  >
    <SelectedIcon />
  </ForwardRef(IconButton)>
  <ForwardRef(Typography)
    color="#F96666"
    variant="h6"
  >
    42
  </ForwardRef(Typography)>
</ForwardRef(Grid)>
`;

exports[`ReviewReaction > should render ReviewReaction type 'L' like correctly when defaultPressed=false  1`] = `
<ForwardRef(Grid)
  alignItems="center"
  direction="row"
  spacing={0.5}
>
  <ForwardRef(IconButton)
    size="small"
  >
    <SelectedIcon />
  </ForwardRef(IconButton)>
  <ForwardRef(Typography)
    color="#C9CCE4"
    variant="h6"
  >
    42
  </ForwardRef(Typography)>
</ForwardRef(Grid)>
`;

exports[`ReviewReaction > should render ReviewReaction type 'L' like correctly when defaultPressed=true  1`] = `
<ForwardRef(Grid)
  alignItems="center"
  direction="row"
  spacing={0.5}
>
  <ForwardRef(IconButton)
    size="small"
  >
    <SelectedIcon />
  </ForwardRef(IconButton)>
  <ForwardRef(Typography)
    color="#6CD62B"
    variant="h6"
  >
    42
  </ForwardRef(Typography)>
</ForwardRef(Grid)>
`;
